test/plugin_functional/plugins/elasticsearch_client_plugin/server/plugin.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Plugin, CoreSetup, CoreStart, ICustomClusterClient } from 'src/core/server';
+
+export class ElasticsearchClientPlugin implements Plugin {
+  private client?: ICustomClusterClient;
+  public setup(core: CoreSetup) {
+    const router = core.http.createRouter();
+    router.get(
+      { path: '/api/elasticsearch_client_plugin/context/ping', validate: false },
+      async (context, req, res) => {
+        const { body } = await context.core.elasticsearch.client.asInternalUser.ping();
```

Comment:
Should we create a x-pack function test that leverages security + `client.asCurrentUser`?